corehq/util/timezones/conversions.py

```
@@ -1,15 +1,35 @@
+import logging
 from django.utils.encoding import smart_str
 import pytz
 from corehq.const import USER_DATETIME_FORMAT
+from corehq.toggles import USE_NEW_TIMEZONE_BEHAVIOR
 from corehq.util.dates import safe_strftime
 from corehq.util.soft_assert import soft_assert
+from corehq.util.view_utils import get_request
 
 
-# The timezone data migration happening some time in Apr-May 2015
-# will shift all phone times (form.timeEnd, case.modified_on, etc.) to UTC
-# so functions that deal with converting to or from phone times
-# use this variable to decide what type of timezone conversion is necessary
-TIMEZONE_DATA_MIGRATION_COMPLETE = False
+def get_timezone_data_migration_complete():
+    """
+    The timezone data migration happening some time in Apr-May 2015
+    will shift all phone times (form.timeEnd, case.modified_on, etc.) to UTC
+    so functions that deal with converting to or from phone times
+    use this function to decide what type of timezone conversion is necessary
+
+    """
+    _default = False
+    request = get_request()
+    if not request or not request.domain:
```

will this fail hard if request doesn't have a domain attribute?